crypto/codec/proto.go

```diff
@@ -25,3 +28,27 @@ func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 	registry.RegisterImplementations((*cryptotypes.PubKey)(nil), &secp256k1.PubKey{})
 	registry.RegisterImplementations((*cryptotypes.PubKey)(nil), &multisig.LegacyAminoPubKey{})
 }
+
+// PubKeyToProto takes a crypto.Pubkey and sets it into tendermint's public key oneof
+// This is only used when sending information about validators to Tendermint (validator updates)
+func PubKeyToProto(k tmcrypto.PubKey) (protocrypto.PublicKey, error) {
```

Comment:
```suggestion
func PubKeyToProto(k cryptotypes.PubKey) (protocrypto.PublicKey, error) {
```

So all in all:
- in the SDK, we manipulate `cryptotypes.PubKey` (our own PubKey)
- whenever we need to use a TM interface (validator updates, `tmtypes.NewValidator`...), we call this function to convert to `protocrypto.PublicKey`

In this case, the SDK should never have to use `tmcrypto.PubKey`, is that correct?